release-notes/swan-lake-2201.1.0-release-note.md

```diff
@@ -30,9 +30,386 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+    string c = float:toExpString(f1, 16); 
+    io:println(c); // -4.5362123339999998e+4
+    string d = float:toExpString(f1, 2); 
+    io:println(d); // -4.54e+4
+}
+```
+
+###### New `lang.string:padStart()`, `lang.string:padEnd()`, and `lang.string:padZero()` functions
+
+`padStart()`, `padEnd()`, and `padZero()` functions have been introduced to add padding in strings. `padStart` adds padding to the start of a string. `padEnd` adds padding to the end of a string. `padZero` pads a string with zeros.
+
+```ballerina
+public function main() {
+    io:println("abc".padStart(5, "#")); // "##abc"
+    io:println("abc".padStart(5)); // "  abc"
+
+    io:println("abc".padEnd(5, "#")); // "abc##"
+    io:println("abc".padEnd(5)); // "abc  "
+
+    io:println("123".padZero(5)); // "00123"
+    io:println("123".padZero(5, "#")); // "##123"
+}
+```
+
+#### Improvements
+
+##### Disallow inferring array sizes from contexts that are not permitted
+
+Inferring array length from the context is restricted to support only with list constructors. Further, only the first dimension can be inferred from the context in multidimensional arrays.
+
+```ballerina
+int[*] x1 = [1, 2]; // Supported.
+
+int[2] y = [1, 2];
+int[*] x2 = y; // Not supported. Required a list constructor to infer            the array length.
+
+int[*][2] x3 = [[1, 2], [1, 2]]; // Supported.
+int[*][*] x4 = [[1, 2], [1, 2]]; // Not supported. Only the first dimension can be inferred from the context.
+```
+
+##### Add annotation attachments to the BIR
+
+
+
+##### Revamp round langlib function for float
+
+The method signature is changed to have an extra argument `fractionDigits`, where the user can choose the number of fraction digits of the rounded result. When `fractionDigits` is zero, the method rounds to an integer.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    io:println(555.545.round(2)); // 555.54
+    io:println(555.545.round(3)); // 555.545
+    io:println(555.545.round(4)); // 555.545
+}
+```
+
+##### `lang.decimal:round` method signature is improved
+
+The method signature is changed to have an extra argument `fractionDigits`, where the user can choose the number of fraction digits of the rounded result. When `fractionDigits` is zero, the method rounds to an integer.
+
+```ballerina
+import ballerina/io;
+import ballerina/lang.'decimal as decimals;
+
+public function main() {
+    io:println(5.55.round(1));	// 5.6
+    decimal x = 5.55;
+    io:println(decimals:round(x));	// 6
+    io:println(decimals:round(5.55, fractionDigits = 0));	// 6
+    io:println(decimals:round(5.5565, fractionDigits = 3));	// 5.556
+}
+```
+
+##### An unreachable panic statement now does not result in a compilation error.
+
+```ballerina
+function fn() returns string {
+    int|string a = 10;
+
+    if a is int {
+        return "INT";
+    } else {
+        return "STRING";
+    }
+
+    panic error("Not Reached!"); // unreachable, but not an error.
+}
+```
+
```

Comment:
We need to add `error:Cloneable` changes
https://github.com/gimantha/ballerina-release/pull/1